sdk/apimanagement/Azure.ResourceManager.ApiManagement/tests/Scenario/ApiManagementServiceCollectionTests.cs

```diff
@@ -0,0 +1,101 @@
+            var apiName = Recording.GenerateAssetName("testapi-");
+            var data = new ApiManagementServiceData(AzureLocation.EastUS, new ApiManagementServiceSkuProperties(ApiManagementServiceSkuType.Developer, 1), "Sample@Sample.com", "sample")
+            {
+                Identity = new ManagedServiceIdentity(ManagedServiceIdentityType.SystemAssigned)
+            };
+            var apiManagementService = (await collection.CreateOrUpdateAsync(WaitUntil.Completed, apiName, data)).Value;
+            Assert.AreEqual(apiManagementService.Data.Name, apiName);
+
+            // TagOperation
+            await apiManagementService.AddTagAsync("testkey", "testvalue");
+            apiManagementService = (await apiManagementService.GetAsync()).Value;
+            Assert.AreEqual(apiManagementService.Data.Tags.FirstOrDefault().Key, "testkey");
+            Assert.AreEqual(apiManagementService.Data.Tags.FirstOrDefault().Value, "testvalue");
+
+            var tags = new Dictionary<string, string>() { { "newkey", "newvalue" } };
+            await apiManagementService.SetTagsAsync(tags);
+            apiManagementService = (await apiManagementService.GetAsync()).Value;
+            Assert.AreEqual(apiManagementService.Data.Tags.FirstOrDefault().Key, "newkey");
+            Assert.AreEqual(apiManagementService.Data.Tags.FirstOrDefault().Value, "newvalue");
+
+            await apiManagementService.RemoveTagAsync("newkey");
+            apiManagementService = (await apiManagementService.GetAsync()).Value;
+            Assert.AreEqual(apiManagementService.Data.Tags.Count, 0);
+
+            // Update
+            var patch = new ApiManagementServicePatch() { Tags = { { "newkey", "newvalue" } } };
+            var updated = await apiManagementService.UpdateAsync(WaitUntil.Completed, patch);
+            Assert.AreEqual(updated.Value.Data.Tags.FirstOrDefault().Key, "newkey");
+
+            // Delete
+            await apiManagementService.DeleteAsync(WaitUntil.Completed);
+        }
+
+        [Test]
+        public async Task Get()
+        {
+            // Please create the resource first.
+            var resourceGroup = await DefaultSubscription.GetResourceGroups().GetAsync("sdktestrg");
+            var collection = resourceGroup.Value.GetApiManagementServices();
+            var apiManagementService = (await collection.GetAsync("sdktestapi")).Value;
+            Assert.NotNull(apiManagementService.Data.Name);
+        }
+
+        [Test]
+        public async Task GetAll()
+        {
+            // Please create the resource first.
+            var resourceGroup = await DefaultSubscription.GetResourceGroups().GetAsync("sdktestrg");
+            var collection = resourceGroup.Value.GetApiManagementServices();
+            var apiManagementServices = await collection.GetAllAsync().ToEnumerableAsync();
+            Assert.GreaterOrEqual(apiManagementServices.Count, 1);
```

Comment:
I wonder why this could pass? We did not even create one instance inside this test case.
The test cases should pass standalone, the result of one test should not depend on the fact of whether we have run other test cases.